WordPress/gutenberg
# Add option to exclude current post from query block

## What?
This PR adds the ability to exclude the current post from the query loop block

Fixes https://github.com/WordPress/gutenberg/issues/42810

## Why?
This is a commonly requested feature by users.

## How?
This is achieved by adding a new attribute to the query block - `excludeCurrent`.

This approach means we can avoid storing specific post IDs in block attributes, meaning if the query loop is copy/pasted between posts/pages or forms part of a pattern, it will behave appropriately.

A toggle control is rendered for this attribute in the block sidebar.

The new attribute is read both in the post template block JS and PHP query generation, to dynamically exclude the current post from the query within the block editor and on the frontend.

Note - the PHP side will require a change to WordPress core in the `build_query_vars_from_query_block` function, I've used a filter here to demonstrate the functionality.

## Testing Instructions
1. Insert a query block into a post/page
2. Toggle the "Exclude current post" option
3. The current post should no longer be visible in the query block in the editor
4. The current post should no longer be visible in the query block on the frontend

### Testing Instructions for Keyboard
The additional UI component uses the existing `ToggleControl` so is keyboard accessible.

## Screenshots or screencast 

https://github.com/user-attachments/assets/05181a9d-8886-4c37-87c5-08be6ce9ad2c
